src/plugins/github/fetchGithubRepo.js

```diff
@@ -76,6 +79,23 @@ export default async function fetchGithubRepo(
   return ((mirror.extract(resolvedId): any): Repository);
 }
 
+// GitHub object IDs are urlsafe-base64-encoded strings that decode to
+// ASCII strings of the form "123:Typename4567[...]", where the "123"
+// numbers are a function only of the typename and the "4567" numbers
+// are the object's database ID, and the "[...]" is either empty or a
+// further section like ":commithash" for commits.
+//
+// See tests for `_guessTypename` for some example object IDs.
+const GITHUB_ID_TYPENAME_PATTERN = /^[0-9]*:([A-Za-z0-9_-]*[A-Za-z_-])[0-9]+(?:[^A-Za-z0-9_-].*)?$/;
+
+export function _guessTypename(
+  objectId: Schema.ObjectId
+): Schema.Typename | null {
+  const decodedId = new Buffer(objectId, "base64").toString("utf-8");
+  const match = decodedId.match(GITHUB_ID_TYPENAME_PATTERN);
+  return match ? match[1] : null;
```

Comment:
Should we check `match[1]` as an enum against expected types first? If one day this starts outputting `BigFurryMonster` as a type we know GitHub has possibly changed the opaque ID format or introduced a new type. Which just so happens to match our regex.

Reply:
If GitHub starts sending a new type here, we’ll certainly want to know
about it, because it means that our local schema is wrong and we’ll do
the wrong thing with the resulting data. We could in principle check the
type against a set of _all_ GitHub’s declared types, but this is large—

```
$ ghquery '{ __schema { types { name } } }' |
> jq '.data.__schema.types[].name' | sort -u | wc -l
740
```

—and may change frequently, and I don’t see a strong reason to do so.